pkg/console/operator/operator.go

```diff
@@ -191,15 +186,8 @@ func (c *consoleOperator) Sync(obj metav1.Object) error {
 	case v311_to_401.BetweenOrEmpty(currentActualVersion):
 		outConfig, configChanged, err = sync_v400(c, outConfig)
 		errs = append(errs, err)
-		if err == nil {
```

Comment:
here I wasnt really sure what to do with setting these status fields, which are not in `v1`, so im nuking them

Reply:
We can trim this down significantly.  I did it once before:

```golang

func (c *consoleOperator) Sync(obj metav1.Object) error {
	startTime := time.Now()
	logrus.Infof("started syncing operator %q (%v)", obj.GetName(), startTime)
	defer logrus.Infof("finished syncing operator %q (%v) \n\n", obj.GetName(), time.Since(startTime))

	operatorConfig := obj.(*consolev1.Console)

	if err := c.handleSync(operatorConfig); err != nil {
		return err
	}
	return nil
}

func (c *consoleOperator) handleSync(config *consolev1.Console) error {

	switch config.Spec.ManagementState {
	case operatorv1.Managed:
		logrus.Println("console is in a managed state.")
		// handled below
	case operatorv1.Unmanaged:
		logrus.Println("console is in an unmanaged state.")
		return nil
		// take a look @ https://github.com/openshift/service-serving-cert-signer/blob/master/pkg/operator/operator.go#L86
	case operatorv1.Removed:
		logrus.Println("console has been removed.")
		return c.deleteAllResources(config)
	default:
		// TODO should update status
		return fmt.Errorf("unknown state: %v", config.Spec.ManagementState)
	}

	// do we need the if(configChanged) update bits?
	outConfig, configChanged, err := sync_v400(c, config)
	if err != nil {
		return err
	}
	if configChanged {
		// TODO: this should do better apply logic or similar, maybe use SetStatusFromAvailability
		_, err = c.operatorClient.Update(outConfig)
		if err != nil {
			return err
		}
	}

	return nil
}

```

Reply:
See here, a previous pass at moving to the `v1`, before things changed: https://github.com/openshift/console-operator/pull/103/files#diff-69ecbb024dab972c8729678abdf9c0faR132